app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -0,0 +1,83 @@
+module Katello
+  class Api::V2::HostSubscriptionsController < Katello::Api::V2::ApiController
+    before_filter :find_host
+    before_filter :check_subscriptions, :only => [:add_subscriptions, :remove_subscriptions]
+
+    api :GET, "/hosts/:host_id/subscriptions", N_("List a host's subscriptions")
+    param :host_id, Integer, :desc => N_("Id of the host"), :required => true
+    def index
```

Comment:
> So we used to do this and IMO it made this controller even more confusing.

I'm still coding according to this design. If there's a new or different way, maybe we could suss it out. My concern is that the old API follows a consistent design both for developers and users.

> And the truth is that a list of attached subscriptions to object X is a different 'object' than a subscription.

This makes sense to me although this 'object' is not entirely clear from the code. I suppose your suggestion would be ok in this case as [we do something similar for `/content_views/8/puppet_modules`](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb). Would rails allow you to route `/subscriptions?host_id=8` to HostSubscriptions#index somehow? This would be nice to have if possible.